cmd/zrepl/zrepl.c

```diff
@@ -331,6 +345,8 @@ uzfs_zvol_io_ack_sender(void *arg)
 		while (1) {
 			if ((zinfo->state == ZVOL_INFO_STATE_OFFLINE) ||
 			    (zinfo->conn_closed == B_TRUE)) {
+				(void) pthread_mutex_unlock(
+				    &zinfo->zinfo_mutex);
```

Comment:
I think we cannot drop the mutex here. The condition for exiting must be checked together with setting ack_sender_created to false, otherwise we risk a race condition with another thread not creating ack thread because it thinks there is one while the existing ack thread is exiting.

Reply:
Done